### DISCUSSION TITLE
Behold Kane's Boustrophedonical Hexagon LED Board!

### DISCUSSION THREAD

**Discussion by davepl** [PlummersSoftwareLLC/NightDriverStrip#325] [Announcements]
Kane kindly sent me one of his hex boards to experiment with - I got it wired up today, as pictured.  By the way, I just discovered those WAGO wire nuts and holy cow do I love them.  Run, do not walk, to get some.

![7A241C17-C9BC-43F7-B5F2-C4F122ACF458](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/3698967/94d58361-a91a-4512-ad31-f4b768c267d7)

I made a new HEXAGON project which is basically a copy of LEDSTRIP_FEATHER_S3, since that's the board I'm using.

![51334676-95EC-4AFF-95EC-61A0FCE4E117](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/3698967/c4eec5fc-30d2-405a-b459-8c5702071ba9)

Fills and so on all work, but now I've got to think up some way to draw to a hex in interesting ways!

**Comment by rbergen** [comment]
That sure looks nice! I'm not worried about you figuring out visually interesting things to do with it. I mean, 6 corners and 6 edges give quite some "hooks" to attach visualizations to. I guess I'll find out what you come up with in (a) future PR(s)! :)

**Comment by davepl** [comment]
Here's my first thinking on it... now to come up with an effect that uses it!

![2023-06-12_21-45-54](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/3698967/4a7a775b-a611-4687-a4fa-a22dc4691022)

**Comment by davepl** [comment]
That seems to work!  I wrote a HexOuterRing effect to draw just the outer ring, and it calcs everything properly and works.  I did a video effect too but it's too big to upload here...

![D4DBACBE-C857-4023-9D55-3547FE939EB9](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/3698967/2201fffb-3b6b-4a63-96c3-ca5c1fee94b0)

**Reply by selfsilent** [reply]
![iap_640x640 3845891446_3f1s78o9](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/49343884/26c5272d-9c82-47bc-a848-3f866baf25eb)
The outline suddenly reminded me of the "Cube-puter" that I made for BigClive. The edges were static and the coloured faces flashed.

**Reply by rbergen** [reply]
Haha @davepl, I told you it wouldn't take long. 😄

**Comment by GPSBabelDeveloper** [comment]
Mine should be only a little behind yours, but subject to import delay randomization.  I found a Stanford Hexagon graphics geek's life work on drawing lines within a hexagonal space by transforming the planes into an X/Y/Z space. (I sent Kane the linky links, but don't have them now on my fone.)  It allows things like drawing rings within (not just ont he outer edge of) and lines between arbitrary points in X/Y/Z space that are 120&deg; apart. We could thus do things like projecting otut increasingly large arbitrary rings emitting from a dart landing within the hex and such. I don't know that it's the final frontier, but I think there's a lot of stuff that would be visually interesting to explore.

I'm catching up on a lot, having received my Mesmerizer just ~64hours ago and my first Neopixels < 12 hours ago, but I think the hexy flexy strip thing (he never answered me on a name...) is pretty interesting so I look forward to volleying some code with others soon.

**Comment by selfsilent** [comment]
As an aside, Wago connectors are brilliant. I'm surprised they took so long to make it the US. It didn't take long for them to be the connector of choice over here in Europe. We still get the name wrong though, calling them Way-go instead of Var-go like the Germans intended.

**Comment by stretch911** [comment]
building the atomic lamp, i invested in the wagos....very very handy!
